mmdet/models/detectors/two_stage.py

```diff
@@ -102,13 +102,26 @@ def forward_train(self,
 
         # assign gts and sample proposals
         if self.with_bbox or self.with_mask:
-            assign_results, sampling_results = multi_apply(
-                assign_and_sample,
-                proposal_list,
-                gt_bboxes,
-                gt_bboxes_ignore,
-                gt_labels,
-                cfg=self.train_cfg.rcnn)
+            bbox_assigner = build_assigner(self.train_cfg.rcnn.assigner)
+            bbox_sampler = build_sampler(
+                self.train_cfg.rcnn.sampler,
+                dict(bbox_roi_extractor=self.bbox_roi_extractor,
+                     bbox_head=self.bbox_head))
```

Comment:
It might be better to modify the interface of `build_sampler` and `build_assigner` to `def build_assigner(cfg, **default_args)`, so that we can just call `build_sampler(self.train_cfg.rcnn.sampler, bbox_roi_extractor=self.bbox_roi_extractor, bbox_head=self.bbox_head)` here.